inception-external-search-solr/src/main/java/de/tudarmstadt/ukp/inception/externalsearch/solr/SolrSearchProvider.java

```diff
@@ -0,0 +1,330 @@
+
+
+public class SolrSearchProvider
+        implements ExternalSearchProvider<SolrSearchProviderTraits>
+{
+
+    private static final String DOC_ID_KEY = "id";
+    private static final String DOC_NAME_KEY = "name";
+    private static final String DOC_TITLE_KEY = "title";
+
+    private static final String DOC_URI_KEY = "uri";
+    private static final String DOC_LANGUAGE_KEY = "language";
+    private static final String DOC_TIMESTAMP_KEY = "timestamp";
+
+    /**
+     * Search documents in a Solr repository
+     * @param aRepository Solr repository
+     * @param aTraits Param for client connection
+     * @param aQuery The query that we retrieve from the search bar
+     * @return A list of results that Inception can read and display
+     * @throws SolrException Connection exception and request exception
+     */
+
+    public List<ExternalSearchResult> executeQuery(DocumentRepository aRepository,
+                                                   SolrSearchProviderTraits aTraits, String aQuery)
+            throws IOException
+    {
+        List<ExternalSearchResult> results = new ArrayList<>();
+        //build client
+
+        HttpSolrClient solrClient = makeClient(aTraits);
+        try {
+
+            SolrQuery query = new SolrQuery();
+            solrClient.query(aTraits.getIndexName(), query);
+            try {
+                query = new SolrQuery();
+                query.setParam("qt", aTraits.getSearchPath());
+                solrClient.query(aTraits.getIndexName(), query);
+
+                if (aQuery.isEmpty() | aQuery.equals("*:*"))
+                    aQuery = "*:*";
+                else {
+                    aQuery = aTraits.getDefaultField() + ":" + aQuery;
+                }
+
+                query.setQuery(aQuery);
+
+                //RANDOM ORDER
+                if (aTraits.isRandomOrder()) {
+                    query.setSort("random_" + aTraits.getSeed(), SolrQuery.ORDER.asc);
+                }
+
+                query.addField("*");
+                query.addField("score");
+
+                query.setHighlight(true);
+                query.addHighlightField(aTraits.getDefaultField());
+
+                query.setRows(aTraits.getResultSize());
+                query.setParam("collection", aTraits.getIndexName());
+
+
+                //RESPONSE
+                QueryResponse response = new QueryResponse();
+                try {
+                    response = solrClient.query(query);
+                    SolrDocumentList documents = response.getResults();
+
+                    for (SolrDocument document : documents) {
+                        ExternalSearchResult result = new ExternalSearchResult(aRepository,
+                                aTraits.getIndexName(), (String) document.getFirstValue("id"));
+
+                        if (!aTraits.isRandomOrder()) {
+                            double d = (float) document.getFirstValue("score");
```

Comment:
Please use constants instead of magic strings - applies to other uses of string literals as well.